### ISSUE TITLE
4-Numpy material structure

### ISSUE THREAD

**Issue by s-weigand** [students-teach-students/python-tools-for-students#23]
Lets make this a kind of checklist, which aspects/functions/methods of numpy should be thought in what order (#4 helps me a lot to write the python material).

**Comment by s-weigand** [comment]
We should at least teach all that is on the [cheatsheet](https://s3.amazonaws.com/assets.datacamp.com/blog_assets/Numpy_Python_Cheat_Sheet.pdf).
Besides file IO maybe, since we will do that in the pandas chapter.
Where numpy can shine in file IO, is reading binary formats with a set spec, but that is too advanced for now I guess.

@redbluee Since you are the author of that chapter, can you make a checklist we can discuss?

**Comment by redbluee** [comment]
Yes i can prepare like a checklist. should be there till this weekend.

**Comment by redbluee** [comment]
# Using NumPy

- [ ] What is NumPy?
- [ ] Data types
- [ ] Creating arrays
- [ ] Arrays attributes
- [ ] Subsetting, Slicing, Indexing
- [ ] Array Manipulation
- [ ] Array Mathematics
  - [ ] Arithmetic operations
  - [ ] Matrices operations
  - [ ] Comparison
  - [ ] Aggregate Functions
- [ ] Sorting arrays
- [ ] Copying arrays
- [ ] Best of all with Matplotlib

**Comment by redbluee** [comment]
@s-weigand

I've updated the contents list, pls check and approve/request changes. Thx!

**Comment by s-weigand** [comment]
Looks nice :smile:

**Comment by s-weigand** [comment]
Maybe reading binary formats would be also interesting i.e. generated from Fortran.
Reading plain text will be explained in the pandas chapter, no need to teach the numpy version.

**Comment by redbluee** [comment]
> Maybe reading binary formats would be also interesting i.e. generated from Fortran.
> Reading plain text will be explained in the pandas chapter, no need to teach the numpy version.

We can add this as a topic before

> * [ ]  Best of all with Matplotlib

, that would be my suggest for C-ordered/FORTRAN-ordered binary files.
